File: app/src/main/java/com/martynaskairys/walltip/ExitAppActivity.java
package com.martynaskairys.walltip;

import android.app.AlarmManager;
import android.app.Dialog;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Collections;
import java.util.HashSet;
import java.util.Random;
import java.util.Set;


public class ExitAppActivity extends AppCompatActivity {

	public static final String CHOSEN_FOLDER_URLS = "chosen_folder_urls";
	public static final String STANDARD = "standard";
	private PendingIntent pendingIntent;
	private String[] imageUrls;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_exit_app);

        final Drawable upArrow = ContextCompat.getDrawable(this, R.drawable.abc_ic_ab_back_mtrl_am_alpha);
        upArrow.setColorFilter(ContextCompat.getColor(this, R.color.primary), PorterDuff.Mode.SRC_ATOP);
        getSupportActionBar().setHomeAsUpIndicator(upArrow);


        setTitle(" ");
        getRandomPictureUrl();

         /* Retrieve a PendingIntent that will perform a broadcast */
        Intent alarmIntent = new Intent(ExitAppActivity.this, AlarmReceiver.class);
        pendingIntent = PendingIntent.getBroadcast(ExitAppActivity.this, 0, alarmIntent, 0);

        setWallpapersToWork();


        findViewById(R.id.buttonExitApp).setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				Intent intent = new Intent(Intent.ACTION_MAIN);
				intent.addCategory(Intent.CATEGORY_HOME);
				intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
				startActivity(intent);
				Toast.makeText(ExitAppActivity.this, R.string.exit_app_button_message, Toast.LENGTH_LONG).show();
			}
		});

        saveUrls();

    }

	private void saveUrls() {
		Set<String> urlsSet = new HashSet<>();
		Collections.addAll(urlsSet, imageUrls);

		SharedPreferences preferences = getSharedPreferences(STANDARD, Context.MODE_PRIVATE);
		SharedPreferences.Editor edit = preferences.edit();
		edit.putStringSet(CHOSEN_FOLDER_URLS, urlsSet);
		edit.apply();
	}

	public String getRandomPictureUrl() {


        imageUrls = getIntent().getStringArrayExtra("images");
        Random randomGenerator = new Random();
        int randomNumber = randomGenerator.nextInt(imageUrls.length);
        return imageUrls[randomNumber];
    }

    public void setWallpapersToWork() {

        AlarmManager manager = (AlarmManager) getSystemService(Context.ALARM_SERVICE);
        int interval = 1000*60*60*24;
        manager.setRepeating(AlarmManager.RTC_WAKEUP, System.currentTimeMillis(), interval, pendingIntent);

        Intent intent = new Intent();
        intent.setAction("AlarmReceiver");
        sendBroadcast(intent);

    }



}

File: app/src/main/java/com/martynaskairys/walltip/ExitAppActivity.kt
package com.martynaskairys.walltip

import android.app.AlarmManager
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.graphics.PorterDuff
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import com.martynaskairys.walltip.ThumbnailActivity.FOLDER_INDEX
import com.martynaskairys.walltip.ThumbnailActivity.IMAGES
import com.martynaskairys.walltip.images.ImageStorageImpl
import com.martynaskairys.walltip.images.ImageStorageManager


class ExitAppActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_exit_app)
        showArrow()
        setupExitButton()

        saveFolderIndexAndUrls()
        changeWallpaperNow()
        scheduleRegularWallpaperUpdates()
    }

    private fun showArrow() {
        val upArrow = ContextCompat.getDrawable(this, R.drawable.ic_arrow_back_white_24dp)
        upArrow.setColorFilter(ContextCompat.getColor(this, R.color.primary), PorterDuff.Mode.SRC_ATOP)
        supportActionBar!!.setHomeAsUpIndicator(upArrow)
    }

    private fun setupExitButton() {
        findViewById(R.id.buttonExitApp)!!.setOnClickListener {
            val intent = Intent(this@ExitAppActivity, LauncherInfoActivity::class.java)
            startActivity(intent)
        }
    }

    private fun saveFolderIndexAndUrls() {
        val imageUrls = intent.getStringArrayExtra(IMAGES)
        val folderIndex = intent.getIntExtra(FOLDER_INDEX, -1)

        if (folderIndex == -1) throw IllegalArgumentException("folder index not found")
        if (imageUrls.isEmpty()) throw IllegalArgumentException("ExitAppActivity received empty urls list: " + imageUrls)

        val imageStorageManager = ImageStorageManager(ImageStorageImpl(applicationContext))
        imageStorageManager.saveUserChosenUrls(imageUrls)
        imageStorageManager.saveUserChosenFolderIndex(folderIndex)
    }

    private fun changeWallpaperNow() {
        val intent = Intent(this, WallpaperService::class.java)
        startService(intent)
    }

    fun scheduleRegularWallpaperUpdates() {

        val manager = getSystemService(Context.ALARM_SERVICE) as AlarmManager
//        val dayInMillis: Long = 1000 * 60
        val dayInMillis: Long = 1000 * 60 * 60 * 24
        val firstTrigger = System.currentTimeMillis() + dayInMillis
        manager.setRepeating(AlarmManager.RTC_WAKEUP, firstTrigger, dayInMillis, getWallpaperPendingIntent())

        val intent = Intent()
        intent.action = "WallpaperServiceReceiver"
        sendBroadcast(intent)
    }

    private fun getWallpaperPendingIntent(): PendingIntent? {
        /* Retrieve a PendingIntent that will perform a broadcast */
        val wallpaperReceiverIntent = Intent(this@ExitAppActivity, WallpaperServiceReceiver::class.java)
        val wallpaperReceiverPendingIntent = PendingIntent.getBroadcast(this@ExitAppActivity, 0,
                wallpaperReceiverIntent, 0)
        return wallpaperReceiverPendingIntent
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                return true
            }
        }

        return super.onOptionsItemSelected(item)
    }
}



